lib/sprinkler/devices/watering_policy.ex

```diff
@@ -0,0 +1,59 @@
+defmodule Sprinkler.Devices.WateringPolicy do
+  # TODO: check this values
+  @soil_moisture_wet 70
+  @soil_moisture_medium_wet 50
+  @ambience_temperature_high 25
+  @humidity_high 50
+
+  def should_irrigate(moisture_input, temperature_input, humidity_input) do
+    unless soil_moisture_wet?(moisture_input) do
+      when_soil_moisture_not_wet(moisture_input, temperature_input, humidity_input)
+    end
```

Comment:
I think this could be written in an easier way, following all those private methods is not easy. Consider this:
```elixir
def irrigation_amount(%GardenReading{moisture: moisture_input, temperature: temperature_input, humidity: humidity_input}) do
  soil_moisture = soil_moisture_level(moisture_input)
  temperature = temperature_level(temperature_input)
  humidity = humidity_level(humidity_input)

  case {soil_moisture, temperature, humidity} do
    # this illustrates the outer left branch on the graph from this PR's description
    when {:medium, :high, _} -> :water_medium
    # rest of the cases
  end
end
```

Reply:
I agree, we could encapsulate all of this in just one method.